clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeClientImpl.java

```diff
@@ -140,6 +145,13 @@ public static ManagedChannel buildChannel(final ZeebeClientConfiguration config)
     channelBuilder.keepAliveTime(config.getKeepAlive().toMillis(), TimeUnit.MILLISECONDS);
     channelBuilder.userAgent("zeebe-client-java/" + VersionUtil.getVersion());
     channelBuilder.maxInboundMessageSize(config.getMaxMessageSize());
+
+    final Map<String, Object> serviceConfig = defaultServiceConfig();
+    if (!serviceConfig.isEmpty()) {
+      channelBuilder.defaultServiceConfig(serviceConfig);
+      channelBuilder.enableRetry();
+    }
```

Comment:
🔧 I think this change would break user space because users don't have the option to override this config yet right? We could consider hiding this behind a feature flag until they can override it.

Reply:
Hm, it's a breaking change on newer versions as it enables retrying by default, as opposed to no retries. As the (broken) tests show at the moment, this can change some of the errors returned.

We could have it opt-in :+1: The main issue is it would be good to have it enabled by default at some point, so where would be the cut off?

Reply:
This is one of those questions that leads to heated debates in other projects. In theory, it could only be done in a major release. 

In practice, I'm fine with such a change in a minor version under the condition that users can override the config entirely with configuration options accompanied by some helpful documentation in the upgrade guide.

Reply:
As I wrote down, I think it's better to keep it opt-in initially, but have be our default in our own tests (QA, E2E, etc.)